public/TCP/IMC.js

```diff
@@ -0,0 +1,514 @@
+const encode = {
+  estimatedState: encodeEstimatedState,
+  entityState: encodeEntityState,
+  desiredControl: encodeDesiredControl,
+  lowLevelControlManeuver: {
+    desiredHeading: encodeLowLevelControlManeuverDesiredHeading,
+    desiredZ: encodeLowLevelControlManeuverDesiredZ,
+  },
+  goTo: encodeGoTo,
+  netFollow: encodeNetFollow,
+  customNetFollow: encodeCustomNetFollowState,
+};
+
+function decode(buf) {
+  let id = buf.readUInt16BE(0);
+  return decodeImc(buf, idToMessageMetadata[id]);
+}
+
+const datatypes = {
+  uint_8t: {
+    name: 'uint_8t',
+    length: 1,
+  },
+  uint_16t: {
+    name: 'uint_16t',
+    length: 2,
+  },
+  uint_32t: {
+    name: 'uint_16t',
+    length: 4,
+  },
+  fp32_t: {
+    name: 'fp32_t',
+    length: 4,
+  },
+  fp64_t: {
+    name: 'fp64_t',
+    length: 8,
+  },
+  bitfield: {
+    name: 'bitfield',
+    length: 1,
+  },
+  recursive: {
+    name: 'recursive',
+  },
+};
+
+function encodeIMC(imcMessage, imcMessageMetadata) {
+  // Check if assumed length is corret
+  if (
+    imcMessageMetadata.length !==
+    imcMessageMetadata.id.datatype.length +
+      lenImcMessage(imcMessageMetadata.message)
+  ) {
+    console.log(
+      `WARNING: Length not equal to calculated length. Assumed length: ${
+        imcMessageMetadata.length
+      } calculated length: ${imcMessageMetadata.id.datatype.length +
+        lenImcMessage(imcMessageMetadata.message)}`,
+    );
+  }
+
+  let buf = Buffer.alloc(imcMessageMetadata.length);
+
+  // Encode ID
+  buf.writeUInt16BE(imcMessageMetadata.id.value);
+  let offset = imcMessageMetadata.id.datatype.length;
```

Comment:
const

Reply:
This cannot be `const`. It changes in the map function